docs/source/RunningTestsRemotely.rst

```diff
@@ -233,3 +231,18 @@ below::
 
 By doing that, both `MYVAR1` and `MYVAR2` will be available in remote
 environment.
+
+Known Issues
+============
+
+Given the modular architecture of Avocado, the fact that the ``remote``
+feature is a plugin and also the fact that the plugins are engaged in no
+particular order, other plugins will not have the information that we
+are in a remote execution. As consequence, plugins that look for local
+resources that are available only remotely can fail. That's the case of
+the so called ``multiplex`` plugin. If you're using the multiplex plugin
+(``-m`` or ``--mux-yaml``) options in addition to the remote plugin (or
+any derived plugin, like ``vm`` or ``docker``), the multiplex files must
+exist locally in the provided path. Notice the multiplex files must be
+also available remotely in the provided path, since we don't copy files
+for remote executions.
```

Comment:
Actually this is even trickier and simpler at the same time. Basically `remote` plugin allows handling of different variants. Not locally, but on a remote machine. This is a knowledge and dependence on a different plugin even though remote plugin could execute multiple variants on a remote machine without the support for multiplexer on the local machine. So instead of trying to fix multiplexer, we should implement different `--remote-mux-yaml` argument used by those remote plugins, which would be independent on the yaml support on the local machine (it'd still require yaml support on the remote machine).